src/DotNetLightning.Core/Channel/CommitmentToLocalExtension.fs

```diff
@@ -0,0 +1,131 @@
+namespace DotNetLightning.Channel
+
+open System
+open NBitcoin
+open NBitcoin.BuilderExtensions
+open DotNetLightning.Utils
+open DotNetLightning.Utils.SeqConsumer
+open DotNetLightning.Crypto
+
+open ResultUtils
+open ResultUtils.Portability
+
+type CommitmentToLocalParameters = {
+    ToSelfDelay: BlockHeightOffset16
+    LocalDelayedPubKey: DelayedPaymentPubKey
+}
+    with
+    static member TryExtractParameters (scriptPubKey: Script): Option<CommitmentToLocalParameters> =
+        let ops =
+            scriptPubKey.ToOps()
+            // we have to collect it into a list and convert back to a seq
+            // because the IEnumerable that NBitcoin gives us is internally
+            // mutable.
+            |> List.ofSeq
+            |> Seq.ofList
```

Comment:
It doesn't work without that conversion back-and-forth. I think the problem was that `SeqConsumer` assumes that you can look at the head of a sequence without changing the sequence, but the `IEnumerable` returned by NBitcoin will pop the head when you do that. ie. this will fail:

```fsharp
let foo = Seq.head sequence
let bar = Seq.head sequence
assert (foo = bar)
```

Reply:
hmm, OK.